lib/py/src/transport/TSocket.py

```diff
@@ -151,6 +151,8 @@ def open(self):
     def read(self, sz):
         try:
             buff = self.handle.recv(sz)
+        except socket.timeout as e:
```

Comment:
starting from python 3.10, this is a _deprecated_ alias to `TimeoutError`: https://docs.python.org/3.12/library/socket.html#socket.timeout

while `TimeoutError` also exists in 3.8 (the earliest version of python still supported): https://docs.python.org/3.8/library/exceptions.html#TimeoutError

so I wonder if we should be more future proof and handle both `socket.timeout` and `TimeoutError` here, and then we can drop `socket.timeout` when 3.10 becomes the earliest version of python supported.

Reply:
modified in https://github.com/apache/thrift/pull/2991